File: code/Project_4/BT5/python/prolems/problem_1_2.py
def is_cycle_bipartite(n):
    """Kiểm tra Cn có hai phía không"""
    return n % 2 == 0

def is_complete_bipartite(n):
    """Kiểm tra Kn có hai phía không"""
    return n <= 2

def main():
    print("Cycle Graph Cn - Bipartite:")
    for n in range(1, 8):
        bipartite = is_cycle_bipartite(n)
        print(f"C{n}: {'Yes' if bipartite else 'No'}")
    
    print("\nComplete Graph Kn - Bipartite:")
    for n in range(1, 6):
        bipartite = is_complete_bipartite(n)
        print(f"K{n}: {'Yes' if bipartite else 'No'}")

if __name__ == "__main__":
    main()

File: code/Project_4/BT5/c++/problems/problem_1_2.cpp
#include <iostream>
using namespace std;

bool is_cycle_bipartite(int n) {
    return n % 2 == 0;
}

bool is_complete_bipartite(int n) {
    return n <= 2;
}

int main() {
    cout << "Cycle Graph Cn - Bipartite:" << endl;
    for (int n = 1; n <= 7; n++) {
        cout << "C" << n << ": " << (is_cycle_bipartite(n) ? "Yes" : "No") << endl;
    }
    
    cout << "\nComplete Graph Kn - Bipartite:" << endl;
    for (int n = 1; n <= 5; n++) {
        cout << "K" << n << ": " << (is_complete_bipartite(n) ? "Yes" : "No") << endl;
    }
    
    return 0;
}